elasticdl/python/ps/parameters.py

```diff
@@ -40,6 +40,26 @@ def set_embedding_param(self, name, indices, values):
             )
         self.embedding_params[name].set(indices, values)
 
+    def check_grad(self, grad):
+        name = grad.name
+        if name in self.non_embedding_params:
+            param_shape = tuple(
+                self.non_embedding_params[name].get_shape().as_list()
+            )
+            if grad.is_indexed_slices():
+                dim0 = tf.math.reduce_max(grad.indices).numpy()
+                dim1 = grad.values.shape[1]
+                if dim0 > param_shape[0] or dim1 != param_shape[1]:
+                    raise ValueError("Gradient %s shape is incompatible", name)
```

Comment:
Shape of the gradient %s is incompatible

Reply:
Done